feat:

SergeSadowsky opened the pull request:
1. Реализуйте все стандартные действия вместе с представлениями (видами) для ресурса вопросов:

    Список вопросов теста должен выводиться на той же странице, что и сам тест
    Используйте фрагменты представлений там, где это уместно
    Для того чтобы метод form_with смог сформировать верный URL используйте вызов form_with model: [@parent, @child]
    В представлениях должны быть ссылки для навигации между страницами приложения


2. Добавьте футер в котором будет выводиться текущий год, имя автора, название проекта (в виде ссылки на GitHub-репозиторий проекта), его краткое описание и ссылка с текстом "Учебный проект в онлайн-школе Thinknetica", ведущая на адрес [thinknetica.com](http://thinknetica.com/)

3. Напишите хэлперы:

    Для заголовка формы создания и редактирования вопроса. Это должен быть один метод question_header который в форме создания вопроса будет выводить Create New <Test Title> Question, а в форме редактирования Edit <Test Title> Question
    Для вывода текущего года в футере
    Для формирования ссылки на GitHub репозиторий: github_url(author, repo)


4. Добавьте в представление списка тестов количество вопросов для каждого теста

uncleserge committed:
feat:
 Added Tetst index view
 Added qustion views
 Created necessary helpers

BubuntuClu commented on app/controllers/questions_controller.rb:
```diff
@@ -1,33 +1,41 @@
 
   def destroy
-    @question.destroy
+    @question.destroy!
```
почему бэнг метод используешь? ты внес уязвимость в код, не добавив обработку на случай если что-то пойдет не так в удалении. бэнг метод выбрасывает исключение

SergeSadowsky replied:
fixed

BubuntuClu commented on app/views/layouts/_footer.html.erb:
```diff
@@ -0,0 +1,7 @@
+<footer>
+  <div>
+    Copyright &copy; <%= current_year %>
+    by <%= github_url author: 'Uncle Serge', repo: 'https://github.com/SergeSadowsky/TestGuru' %>.
```
всегда ставь скобки вокруг параметров методов. без них ты можешь спровоцировать очень трудно находимую ошибку

SergeSadowsky replied:
fixed

BubuntuClu commented on app/views/questions/edit.html.erb:
```diff
@@ -0,0 +1,8 @@
+<%= question_header(@question) %>
```
вьюхи new/edit по сути одинаковые. в них должно быть только `<%= render "form", question: @question %>`.

зачем делать отдельный метод для отображения заголовка по состоянию объекта если мы вызываем код на конкретной вьюхе и знаем что отобразить надо?

SergeSadowsky replied:
fixed

uncleserge committed:
fix:
fixed bang destroy
fixed parenthesis
fixed new/edit views
fixed question.rescue_with_question_not_found

BubuntuClu commented on app/views/questions/edit.html.erb:
```diff
@@ -1,8 +1,7 @@
 <div>
-  <%= link_to "Back to questions", test_questions_path(@question.test) %>
+  <%= link_to "Back to questions", test_questions_path(@question.test) %> |
+  <%= link_to 'Delete', question_path(@question),
```
а нужно ли тут делать кнопку удаления? почему не на списке вопросов?

BubuntuClu replied:
и, кстати, говоря о списке вопросов

>Список вопросов теста должен выводиться на той же странице, что и сам тест

задание не сделано. #index у вопросов можно удалить, так как информация эта должна выводиться в tests#show

SergeSadowsky replied:
> а нужно ли тут делать кнопку удаления? почему не на списке вопросов?
Удаление присутствует и в списке вопросов и в Show вопроса



> и, кстати, говоря о списке вопросов
> 
> > Список вопросов теста должен выводиться на той же странице, что и сам тест
> 
> задание не сделано. #index у вопросов можно удалить, так как информация эта должна выводиться в tests#show

Создал Test.show, удалил Test.questions.index

uncleserge committed:
fix:
created Test.Show instead Test.Questions.Index
